src/main/java/ca/on/oicr/pineryreports/Main.java

```diff
@@ -0,0 +1,149 @@
+      System.exit(1);
+    } catch (HttpResponseException e) {
+      System.err.println("Error fetching data from Pinery");
+      e.printStackTrace();
+      System.exit(2);
+    } catch (IOException e) {
+      System.err.println("Error writing to file");
+      e.printStackTrace();
+      System.exit(3);
+    }
+  }
+  
+  private static CommandLine getCommandLine(String[] args, Options opts, boolean acceptExtras) throws ParseException {
+    CommandLineParser parser = new DefaultParser();
+    return parser.parse(opts, args, !acceptExtras);
+  }
+  
+  private static Options getMainOptions() {
+    Options opts = new Options();
+    
+    opts.addOption(Option.builder("s")
+        .longOpt("source")
+        .required()
+        .hasArg()
+        .argName("pinery-url")
+        .desc("Pinery base URL")
+        .build());
+    opts.addOption(Option.builder("r")
+        .longOpt("report")
+        .required()
+        .hasArg()
+        .argName("name")
+        .desc("Report to generate {stock, gecco}")
+        .build());
+    opts.addOption(Option.builder("f")
+        .longOpt("format")
+        .hasArg()
+        .argName("code")
+        .desc("Output format. Options and default depend on report {csv, pdf}")
+        .build());
+    opts.addOption(Option.builder("o")
+        .hasArg()
+        .argName("filename")
+        .desc("Output file (Default: report.<format-code>")
+        .build());
+    
+    
+    opts.addRequiredOption("s", "pinery-source", true, "Pinery base URL");
+    opts.addRequiredOption("r", "report", true, "Report to generate (Options: stock)");
+    opts.addOption("f", "format", true, "Output format. Options and default depend on report (csv, pdf)");
+    opts.addOption("o", true, "Output file (Default: report.<format>)");
+    
+    return opts;
+  }
+  
+  private static Report getReport(String reportName) throws ParseException {
+    switch (reportName) {
+    case StockReport.REPORT_NAME:
+      return new StockReport();
+    case GeccoReport.REPORT_NAME:
+      return new GeccoReport();
+    default:
+      throw new ParseException("Invalid report requested: " + reportName);
+    }
+  }
+  
+  private static void showHelp(Report report) {
+    HelpFormatter formatter = new HelpFormatter();
+    Options mainOpts = getMainOptions();
+    formatter.printHelp("java -jar pinery-reports.jar -s <pinery-url> -r <report> [-f {pdf|csv}]"
+        + " [-o <filename>] [report-specific options]", mainOpts);
+    
+    if (report != null) {
+      Options reportOpts = new Options();
+      for (Option opt : report.getOptions()) {
+        reportOpts.addOption(opt);
+      }
+      System.out.println("");
+      formatter.printHelp(String.format("Report-specific options (%s)", report.getReportName()), reportOpts);
+    }
+  }
+  
+  // TODO remove me after sonarcloud does its thing
+  public void junkMethodIsJunk(String unusedParam) throws Exception {
+    if (true) {
+      throw new Exception("this is garbage");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)